test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ExportsTest.scala

```diff
@@ -25,8 +25,24 @@ import org.scalajs.testsuite.utils.AssertThrows.assertThrows
 
 class ExportsTest {
 
-  /** The namespace in which top-level exports are stored. */
-  val exportsNamespace = scala.scalajs.runtime.environmentInfo.exportsNamespace
+  /** The namespace in which top-level exports are stored.
+   *
+   *  If we are linking the test suite in `NoModule`, then exports are in the
+   *  global object (technically they're in the global scope, but at least so
+   *  far we can find them in the global object too).
+   *
+   *  If we are linking in `CommonJSModule`, then exports are in the `exports`
+   *  module-global variable, which we can retrieve as if it were in the global
+   *  scope.
+   */
+  val exportsNamespace: js.Dynamic = {
+    if (Platform.isNoModule)
+      org.scalajs.testsuite.utils.JSUtils.globalObject
+    else if (Platform.isCommonJSModule)
+      js.Dynamic.global.exports
+    else
+      ???
```

Comment:
I thought that `NotImplementedError` was the best possible description in this case. I'll write a proper error message.